### DISCUSSION TITLE
Initial setup and use

### DISCUSSION THREAD

**Discussion by snake624** [the-anylogic-company/AnyLogic-Pypeline#47] [Q&A] [answered]
Read through troubleshooting and through the wiki and did not see this one:

I have tried multiple example models as well as a blank new model and get the same errors:

Error during model creation:
Unresolved compilation problems: 
	The type anylogic_jsonifier_library.Jsonifier cannot be resolved. It is indirectly referenced from required .class files
	The method createAsEmbedded() is undefined for the type PyCommunicator
	The method startAsEmbedded() is undefined for the type PyCommunicator
	The method doFinish() is undefined for the type PyCommunicator

java.lang.Error: Unresolved compilation problems: 
	The type anylogic_jsonifier_library.Jsonifier cannot be resolved. It is indirectly referenced from required .class files
	The method createAsEmbedded() is undefined for the type PyCommunicator
	The method startAsEmbedded() is undefined for the type PyCommunicator
	The method doFinish() is undefined for the type PyCommunicator

	at model.Main.<init>(Main.java:1)
	at model.Simulation.createRoot(Simulation.java:155)
	at model.Simulation.createRoot(Simulation.java:1)

package model;

import java.io.Serializable;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Calendar;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Currency;
import java.util.Date;
import java.util.Enumeration;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Hashtable;
import java.util.Iterator;
import java.util.LinkedHashMap;
import java.util.LinkedHashSet;
import java.util.LinkedList;
import java.util.List;
import java.util.ListIterator;
import java.util.Locale;
import java.util.Map;
import java.util.PriorityQueue;
import java.util.Random;
import java.util.Set;
import java.util.SortedMap;
import java.util.SortedSet;
import java.util.Stack;
import java.util.Timer;
import java.util.TreeMap;
import java.util.TreeSet;
import java.util.Vector;
import java.awt.Color;
import java.awt.Font;
import com.anylogic.engine.connectivity.ResultSet;
import com.anylogic.engine.connectivity.Statement;
import com.anylogic.engine.elements.*;
import com.anylogic.engine.markup.Network;
import com.anylogic.engine.Position;
import com.anylogic.engine.markup.PedFlowStatistics;
import com.anylogic.engine.markup.DensityMap;

import static java.lang.Math.*;
import static com.anylogic.engine.UtilitiesArray.*;
import static com.anylogic.engine.UtilitiesCollection.*;
import static com.anylogic.engine.presentation.UtilitiesColor.*;
import static com.anylogic.engine.HyperArray.*;

import com.anylogic.engine.*;
import com.anylogic.engine.analysis.*;
import com.anylogic.engine.connectivity.*;
import com.anylogic.engine.database.*;
import com.anylogic.engine.gis.*;
import com.anylogic.engine.markup.*;
import com.anylogic.engine.routing.*;
import com.anylogic.engine.presentation.*;
import com.anylogic.engine.gui.*;

import com.anylogic.libraries.modules.markup_descriptors.*;
import com.anylogic.libraries.pypeline.*;

public class Simulation extends ExperimentSimulation<Main> {
  @AnyLogicInternalCodegenAPI
  public static String[] COMMAND_LINE_ARGUMENTS_xjal = new String[0];
  {
    setCommandLineArguments_xjal( COMMAND_LINE_ARGUMENTS_xjal );
  }
  @AnyLogicInternalCodegenAPI
  private static Map<String, IElementDescriptor> elementDesciptors_xjal = createElementDescriptors( Simulation.class );
  
  @AnyLogicInternalCodegenAPI
  @Override
  public Map<String, IElementDescriptor> getElementDesciptors() {
    return elementDesciptors_xjal;
  }
  // View areas
  @AnyLogicInternalCodegenAPI
  protected static final Font _text_Font = new Font("SansSerif", 0, 24 );
  @AnyLogicInternalCodegenAPI
  protected static final int _text = 1;

  /** Internal constant, shouldn't be accessed by user */
  @AnyLogicInternalCodegenAPI
  protected static final int _SHAPE_NEXT_ID_xjal = 2;

  protected ShapeText text;
  @AnyLogicInternalCodegenAPI
  private void _createPersistentElementsBP0_xjal() {
    text = new ShapeText(
        SHAPE_DRAW_2D, true,50.0, 30.0, 0.0, 0.0, 
        royalBlue,"Model",
        _text_Font, ALIGNMENT_LEFT );

  }

  @AnyLogicInternalCodegenAPI
  private void _createPersistentElementsAP0_xjal() {
  }

  @AnyLogicInternalCodegenAPI
  private void _createPersistentElementsBS0_xjal() {
  }

  protected ShapeTopLevelPresentationGroup presentation;
  protected ShapeModelElementsGroup icon; 

  @Override
  @AnyLogicInternalCodegenAPI
  public ShapeTopLevelPresentationGroup getPresentationShape() {
    return presentation;
  }

  @Override
  @AnyLogicInternalCodegenAPI
  public ShapeModelElementsGroup getModelElementsShape() {
    return icon;
  }

  @Override
  public int getWindowWidth() {
    return 1000;
  }

  @Override
  public int getWindowHeight() {
    return 600;
  }
  
  @Override
  @AnyLogicInternalCodegenAPI
  public void onDestroy_xjal() {
    super.onDestroy_xjal();
  }

  @Override
  @AnyLogicInternalCodegenAPI
  public void initDefaultRandomNumberGenerator(Engine _e) {
    _e.getDefaultRandomGenerator().setSeed( 1 );
  }

  @Override
  @AnyLogicInternalCodegenAPI
  public Main createRoot( Engine engine ) {
    // Create the root object
    return new Main( engine, null, null );
  }
  
  @Override
  @AnyLogicInternalCodegenAPI
  public void setupRootParameters( final Main self, boolean callOnChangeActions ) {
    final Main root = self; // for compatibility
  }
  

  /**
   * Engine setup
   */
  @Override
  @AnyLogicInternalCodegenAPI
  public void setupEngine(Engine engine) {
    engine.setATOL( 1.0E-5 );
    engine.setRTOL( 1.0E-5 );
    engine.setTTOL( 1.0E-5 );
    engine.setHTOL( 0.001 );
    engine.setSolverODE( Engine.SOLVER_ODE_EULER );
    engine.setSolverNAE( Engine.SOLVER_NAE_MODIFIED_NEWTON );
    engine.setSolverDAE( Engine.SOLVER_DAE_RK45_NEWTON );
    engine.setVMethods( 394549 );
    engine.setSimultaneousEventsSelectionMode( Engine.EVENT_SELECTION_LIFO );

    engine.setStartTime( 0.0 );
    engine.setTimeUnit( SECOND );
    engine.setStartDate( toDate( 2023, MAY, 25, 0, 0, 0 ) );
    engine.setRealTimeMode( true );
    engine.setRealTimeScale( 1.0 );
  }

  /**
   * Experiment setup
   */
  @Override
  @AnyLogicInternalCodegenAPI
  public void setup( IExperimentHost _e ) {
    setName( "Model : Simulation" );

    // Static initialization of persistent elements
    _createPersistentElementsBP0_xjal();

    // Dynamic initialization of persistent elements
    _createPersistentElementsAP0_xjal();
    presentation = new ShapeTopLevelPresentationGroup( Simulation.this, true, 0, 0, 0, 0 , text );
    // Dynamic initialization of persistent elements
    _createPersistentElementsBS0_xjal();
    icon = new ShapeModelElementsGroup( Simulation.this, getElementProperty( "model.Simulation.icon", IElementDescriptor.MODEL_ELEMENT_DESCRIPTORS )  );
    _e.setZoomAndPanningEnabled( true );
    _e.setDeveloperPanelEnabled( true );
    _e.setDeveloperPanelVisibleOnStart( false );
	runAndShowRoot_xjal();
  }

}

Any help would be appreciated.

**Comment by t-wolfeadam** [comment]
@snake624 What version of Pypeline are you using? It should be at least v1.8 (go to manage libraries (from the "+" in the Palette) and check in the table.

**Reply by t-wolfeadam** [reply]
If you're on a version less than that, close AnyLogic and replace the "Pypeline.jar" with the latest one from [Releases](https://github.com/t-wolfeadam/AnyLogic-Pypeline/releases)

**Reply by snake624** [reply]
@t-wolfeadam I was using v1.8, but when I reverted back to v1.71 it worked.

**Reply by t-wolfeadam** [reply]
That's weird...are the models from the same version? 
I would maybe try a clean slate. I suggest downloading the latest version and matching examples (literally just released!) -- [here](https://github.com/t-wolfeadam/AnyLogic-Pypeline/releases/tag/v1.9.0)

**Reply by snake624** [reply]
Downloaded the v1.9 and ran into the same problem and errors. Switched back to v1.71 and up and running again.

**Reply by snake624** [reply]
I tried with a brand new Model, Your example models, and a model I have made all in AnyLogic 8 University 8.8.3.

**Comment by t-wolfeadam** [comment, marked as answer]
As described in #51, this was a problem since 1.7.1 -- fixed in [v1.9.1](https://github.com/t-wolfeadam/AnyLogic-Pypeline/releases/tag/v1.9.1)
